### PR TITLE
[RLlib] Add necessary fields to Base Envs, and BaseEnv wrapper classes

### PR DESCRIPTION
**Pull request by avnishn**
This is an initial commit. Adding observation_space
action_space, and a method for retrieving env_ids as
requirements to the BaseEnv Class

Will need to update for all of the other BaseEnv
wrappers

## Why are these changes needed?
Base Envs are missing attributes that are necessary for validating them. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

**Review by gjoliver — COMMENTED** [review]
I like this.
do observation_space(self) and action_space(self) work for all the other wrapper types?

**Comment by avnishn** [comment]
They don't work yet, I am adding them in, which is why I made this a draft PR 😁

**Comment by gjoliver** [comment]
> They don't work yet, I am adding them in, which is why I made this a draft PR 😁

ok, ping after you are done 👍

**Review by gjoliver — COMMENTED** [review]
still looks very nice. like the dict space design for multi-agent env.

just have one question, is it really worth it to have all the sub_environments() function take an additional as_dict parameter?
why not always return a dict?
it's not gonna make big difference when you use it, since you can simply .values() the returned dict.

**Comment by gjoliver** [comment]
looks great to me!!
can you make sure all the tests pass?
we can then merge this.

File: rllib/env/base_env.py [hunk 1]
```diff
@@ -185,12 +186,18 @@ def try_reset(self, env_id: Optional[EnvID] = None
         return None
 
     @PublicAPI
-    def get_sub_environments(self) -> List[EnvType]:
+    def get_sub_environments(
```
Comment (sven1977) [inline rllib/env/base_env.py:189]: Nice!

File: rllib/env/vector_env.py [hunk 2]
```diff
@@ -312,10 +312,44 @@ def try_reset(self, env_id: Optional[EnvID] = None) -> MultiEnvDict:
...
+
+    @property
+    @override(BaseEnv)
+    @PublicAPI
+    def action_space(self) -> gym.Space:
+        return self._action_space
+
+    @staticmethod
+    def _space_contains(space: gym.Space, x: MultiEnvDict) -> bool:
+        """Check if
```
Comment (sven1977) [inline rllib/env/vector_env.py:345]: Could you complete this docstring?

**Review by gjoliver — APPROVED** [review]

**Pull request merged**